### ISSUE TITLE
Redef'd record not resolved if behind unqualified ID

### ISSUE THREAD

**Issue by bbannier** [bbannier/zeek-language-server#2]
We don't resolve some redef'd records. We should resolve at any location marked `^` below.

```zeek
module x;

type A: record {};
redef record A += { c: count &optional; };

function f(a: A) { a$c; }
           ^  ^    ^ ^
```

If the function was declared `function f(a: x::A) { a$c; }` I see the expected behavior.

**Comment by bbannier** [comment]
Once this is fixed the following test should pass:

```rust
#[test]
fn redef_use_in_module_decl_elsewhere() {
    let mut db = TestDatabase::new();
    let uri = Arc::new(Url::from_file_path("/x.zeek").unwrap());

    db.add_prefix("/p");

    db.add_file(
        Arc::new(Url::from_file_path("/p/init-bare.zeek").unwrap()),
        "module GLOBAL;
type connection: record { id: string; };",
    );
    db.add_file(
        Arc::new(Url::from_file_path("/p/ftp/info.zeek").unwrap()),
        "module FTP;
export {
    type Info: record { uid: string &log; };
}
redef record connection += { ftp: Info &optional; };",
    );
    db.add_file(
        Arc::new(Url::from_file_path("/p/ftp/main.zeek").unwrap()),
        "module FTP;
@load ./info
redef record connection += { ftp: Info &optional; };",
    );
    db.add_file(
        Arc::new(Url::from_file_path("/p/init-default.zeek").unwrap()),
        "module FTP;
@load ftp/main",
    );

    db.add_file(
        uri.clone(),
        "module x;
@load init-bare
@load init-default
event connection_established(c: connection) {
    c$ftp$uid;
}",
    );

    let db = db.snapshot();
    let tree = db.parse(uri.clone()).unwrap();
    let source = db.source(uri.clone());

    let ftp = tree
        .root_node()
        .named_descendant_for_position(Position::new(4, 6))
        .unwrap();
    assert_eq!(ftp.utf8_text(source.as_bytes()), Ok("ftp"));
    let r_ftp = db
        .resolve(NodeLocation::from_node(uri.clone(), ftp))
        .unwrap();
    assert_debug_snapshot!(r_ftp);
    assert_debug_snapshot!(db.typ(r_ftp).unwrap());

    // FIXME(bbannier): remove debug stmts.
    assert_debug_snapshot!(dbg!(db.resolve_id(
        Arc::new("connection".to_string()),
        NodeLocation::from_node(uri.clone(), ftp),
    )));
}
```

**Comment by bbannier** [comment]
This has been fixed by 224aeffe35649d33ba348e8e45d6be66b2917cf6 which was released with v0.2.0 right around the time this issue was created.
